Orange/data/table.py

```diff
@@ -2029,10 +2267,31 @@ def _check_inf(array):
 
 def _subarray(arr, rows, cols):
     rows = _optimize_indices(rows, arr.shape[0])
+    if arr.ndim == 1:
+        return arr[rows]
     cols = _optimize_indices(cols, arr.shape[1])
     return arr[_rxc_ix(rows, cols)]
 
 
+# TODO: Pick that one or the one above
```

Comment:
@markotoplak, should `_subarray` avoid copying when possible? Which function do you prefer - the one above or the one below.

The one at the bottom allows unlocking in cases, where the one above would prevent them because it creates a view (and view can't be unlocked). The problem with the function below is just backward compatibility - perhaps somebody expected to get a copy. But I don't think we should care.

Reply:
I like your modified version, thanks!